oddt/fingerprints.py

```diff
@@ -405,15 +409,129 @@ def ECFP(mol, depth=2, size=4096, count_bits=True, sparse=True,
     return mol_hashed
 
 
+def SPLIF(ligand, protein, depth=1, size=4096):
+    """
+        Calculates structural protein-ligand interaction fingerprint (SPLIF),
+        based on http://pubs.acs.org/doi/abs/10.1021/ci500319f.
+
+        Parameters
+        ----------
+        ligand, protein : oddt.toolkit.Molecule object
+                Molecules, which are analysed in order to find interactions.
+        depth : int (deafult = 1)
+            The depth of the fingerprint, i.e. the number of bonds in Morgan
+            algorithm. Note: For ECFP2: depth = 1, ECFP4: depth = 2, etc.
+        size: int (default = 4096)
+            SPLIF is folded to given size.
+
+        Returns
+        -------
+        SPLIF : numpy array
+            SPLIF = [no of atoms, 1], column consists of:
+                Column[0] = index of hashed atoms
+                Column[1] = neigbor's coords of ligand's atom,
+                (5, 3) -> ligand's atom coords and 4 his neigbors
+                Column[2] = neigbor's coords of protein's atom,
+                (5, 3) -> protein's atom coords and 4 his neigbors
+
+    """
+    # removing h
+    protein_dict = protein.atom_dict[protein.atom_dict['atomicnum'] != 1]
+    ligand_dict = ligand.atom_dict[ligand.atom_dict['atomicnum'] != 1]
+
+    protein_atoms, ligand_atoms = close_contacts(
+        protein_dict, ligand_dict, cutoff=4.5)
+    SPLIF = np.zeros((len(ligand_atoms)),
+                     dtype=[('hash', int), ('ligand_coords', np.float32, (5, 3)),
+                            ('protein_coords', np.float32, (5, 3))])
+    for index in range(len(ligand_atoms)):
```

Comment:
What if we use the actual indices instead of position in for.

> for lig_dict, prot_dict in zip(ligand_atoms, protein_atoms)

It even might speed up things.